app/controllers/sites_controller.rb

```diff
@@ -1,8 +1,17 @@
 class SitesController < ApplicationController
-  before_action :authenticate
+  before_action :authenticate, :build_provider
 
   def index
-    @provider = Provider.includes(:sites).find(params[:provider_code]).first
     @sites = @provider.sites.sort_by(&:location_name)
   end
+
+  def edit
+    @site = @provider.sites.find(params[:id]).first
```

Comment:
I don't think this `.find` works...

```
@provider.sites.first
=> #<Site:@attributes={"type"=>"sites", "id"=>"9576475", "code"=>"R", "location_name"=>"One Degree Academy", "address1"=>"Queensway", "address2"=>"Enfield", "address3"=>"", "address4"=>"", "postcode"=>"EN3", "region_code"=>"london"}>
@provider.sites.find(999).first
=> #<Site:@attributes={"type"=>"sites", "id"=>"9576475", "code"=>"R", "location_name"=>"One Degree Academy", "address1"=>"Queensway", "address2"=>"Enfield", "address3"=>"", "address4"=>"", "postcode"=>"EN3", "region_code"=>"london"}>
```

Reply:
Is that in the tests? Cause webmock doesn't care what the ID is, it just returns the factory responses.